unit-test/background/reference-tests/referrer-trimming-tests.js

```diff
@@ -91,16 +93,17 @@ for (const setName of Object.keys(testSets)) {
                         enumerable: true,
                         writeable: true
                     })
-                    FakeDocument.prototype.location = {
-                        url: test.frameURL || test.siteURL
-                    }
                     const orgDocument = globalThis.Document
 
                     // replacing real document and Document with fake ones
                     globalThis.Document = FakeDocument
                     spyOnProperty(document, 'referrer', 'get').and.returnValue(test.referrerValue)
+                    spyOnProperty(document, 'URL', 'get').and.returnValue(test.frameURL || test.siteURL)
 
-                    jsReferrerProtection.init({ referrer: tab.referrer })
+                    const args = getArgumentsObject(1, { url: test.siteURL, frameId: 0 }, test.siteURL, 'abc123')
+                    if (!isFeatureBroken(args, 'referrer')) {
```

Comment:
We're doing this in other tests, it's not ideal as mentioned on MM but changing this to instantiate the whole of C-S-S is large rewrite.

We're essentially mocking the message from the extension which used the `getArgumentsObject` and calling the `isFeatureBroken` method from C-S-S to validate if we should trigger the calling of the method.